nixos/modules/security/pam.nix

```diff
@@ -463,6 +505,9 @@ let
               "session optional ${pkgs.gnome3.gnome-keyring}/lib/security/pam_gnome_keyring.so auto_start"}
           ${optionalString (config.virtualisation.lxc.lxcfs.enable)
                "session optional ${pkgs.lxc}/lib/security/pam_cgfs.so -c all"}
+
+          ${optionalString (cfg.externalCommand != null)
```

Comment:
I wonder if we should move this to a separate file. However there might be a reason we have a monolith here because the order matters?

Reply:
I do not think the order matters.

Reply:
We really shouldn't add more to this monolith IMHO. Please see https://github.com/NixOS/nixpkgs/issues/90640.

Reply:
@datafoo yes, the order does matter. I'm with @flokli on this one, the PAM module needs some real reworking, let's not make it worse.

Reply:
In PAM, order very much matters. Especially when using enterprise solutions such as AD/SSSD etc. @flokli pointed me here from my issue #104346 where having pam_unix.so as required, and before pam_sss.so causes some major porblems. pam_unix.so will always fail a non-local account, so by requiring pam_unix.so all network/SSSD accounts will fail.